dipy/reconst/tests/test_multi_voxel.py

```diff
@@ -171,44 +180,63 @@ def predict(self, S0):
     npt.assert_equal(type(fit), SillyFit)
 
     # Test without a mask
     many_voxels = np.zeros((2, 3, 4, 64))
-    fit = model.fit(many_voxels)
-    expected = np.empty((2, 3, 4))
-    expected[:] = 2.0
-    npt.assert_array_equal(fit.model_attr, expected)
-    expected = np.ones((2, 3, 4, 12))
-    npt.assert_array_equal(fit.odf(unit_icosahedron), expected)
-    npt.assert_equal(fit.directions.shape, (2, 3, 4))
-    S0 = 100.0
-    npt.assert_equal(fit.predict(S0=S0), np.ones(many_voxels.shape) * S0)
-
-    # Test with parallelization (using the "serial" dummy engine)
-    fit = model.fit(many_voxels, engine="serial")
-
-    # If parallelization engines are installed use them to test:
-    if has_joblib:
-        fit = model.fit(many_voxels, engine="joblib")
+    for verbose in [True, False]:
+        fit = model.fit(many_voxels, verbose=verbose, another_kwarg="foo")
+        expected = np.empty((2, 3, 4))
+        expected[:] = 2.0
+        npt.assert_array_equal(fit.model_attr, expected)
+        expected = np.ones((2, 3, 4, 12))
+        npt.assert_array_equal(fit.odf(unit_icosahedron), expected)
+        npt.assert_equal(fit.directions.shape, (2, 3, 4))
+        S0 = 100.0
         npt.assert_equal(fit.predict(S0=S0), np.ones(many_voxels.shape) * S0)
 
-    if has_dask:
-        fit = model.fit(many_voxels, engine="dask")
-        npt.assert_equal(fit.predict(S0=S0), np.ones(many_voxels.shape) * S0)
-
-    if has_ray:
-        fit = model.fit(many_voxels, engine="ray")
-        npt.assert_equal(fit.predict(S0=S0), np.ones(many_voxels.shape) * S0)
+    # Test with parallelization (using the "serial" dummy engine)
+    fit = model.fit(many_voxels, another_kwarg="foo", engine="serial")
+
+    for verbose in [True, False]:
+        # Test with single value kwarg, or sequence of kwarg values
+        for another_kwarg in ["foo", len(many_voxels) * ["foo"]]:
+            # If parallelization engines are installed use them to test:
+            if has_joblib:
+                fit = model.fit(
+                    many_voxels,
+                    verbose=verbose,
+                    another_kwarg=another_kwarg,
+                    engine="joblib",
+                )
+                npt.assert_equal(fit.predict(S0=S0), np.ones(many_voxels.shape) * S0)
+
+            if has_dask:
+                fit = model.fit(
+                    many_voxels,
+                    verbose=verbose,
+                    another_kwarg=another_kwarg,
+                    engine="dask",
+                )
+                npt.assert_equal(fit.predict(S0=S0), np.ones(many_voxels.shape) * S0)
+
+            if has_ray:
+                fit = model.fit(
+                    many_voxels,
+                    verbose=verbose,
+                    another_kwarg=another_kwarg,
+                    engine="ray",
+                )
+                npt.assert_equal(fit.predict(S0=S0), np.ones(many_voxels.shape) * S0)
 
```

Comment:
is it relevant to check if it is faster than `serial` ?  small data so not sure it will be stable. Let me know what you think

Reply:
I don't think we want to test this here. As you say, with this size data, it's not a given that it will always be faster, and may vary depending on other context: platform, hardware, etc.